tt_eager/tt_dnn/op_library/bmm/bmm_op.cpp

```diff
@@ -363,10 +351,18 @@ tuple<uint32_t, uint32_t> get_matmul_subblock_params(const uint32_t per_core_M,
 
 
 // TODO: Only supports sharded matmul for now; infer most matmul params from shard spec
-tt::operations::primary::MatmulProgramConfig get_matmul_program_config(const Tensor &input_tensor_a, const Tensor &input_tensor_b, const MemoryConfig &output_mem_config, std::optional<UnaryWithParam> fused_activation, const bool matmul) {
+tt::operations::primary::MatmulProgramConfig get_matmul_program_config(const Tensor &input_tensor_a, const Tensor &input_tensor_b, const MemoryConfig &output_mem_config, std::optional<UnaryWithParam> fused_activation, const bool matmul, const std::optional<const CoreCoord> core_coord, std::optional<const DeviceComputeKernelConfig> compute_kernel_config) {
     TT_FATAL(input_tensor_a.is_sharded());
-    // TODO: Should we check if grid_size is valid against device->compute_with_storage_grid_size()?
+    bool fp32_dest_acc_en = bmm_op_utils::get_fp32_dest_acc_en(compute_kernel_config);
     auto grid_size = input_tensor_a.shard_spec().value().grid.bounding_box().grid_size();
+    if (core_coord) {
+	auto coord = *core_coord;
+	auto storage_grid_size = input_tensor_a.device()->compute_with_storage_grid_size();
+	if (coord.x <= storage_grid_size.x && coord.y <= storage_grid_size.y) {
+	    grid_size.x = coord.x;
+	    grid_size.y = coord.y;
+	}
+    }
```

Comment:
- I think we need a better name than just `core_coord`
- I think `core_coord.value()` is better than `*core_coord`?
- Do we want to only override the grid_size only if both x and y are valid or override them individually if they are within bounds?
 
At a higher level, when input_a is sharded, we always assume the compute grid is the shard_grid (ie. `bounding_box().grid_size()`). I think we should not allow override for now until we separate the compute grid from the input shard grid. This requires:
- changing how we create sharded output tensors (since this will now be based on the user provided grid)
- changing micro-ops to support arbitrary compute grid

Reply:
I'll use `user_core_coord` and replace the if statement with a TODO. 

In the future, I would say we'd want to override the grid size only if both x and y are valid, and print a warning if they are not.